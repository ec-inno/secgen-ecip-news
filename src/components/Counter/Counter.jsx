import React from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';

import Icon from '../Icon';

const Counter = ({ requests, registered, successful }) => {
  const { t } = useTranslation();

  return (
    <p className="ecl-u-type-paragraph ecl-u-mt-l ecl-u-d-flex ecl-u-justify-content-between">
      <span className="ecl-u-d-flex ecl-u-align-items-center">
        <Icon className="ecl-u-mr-xs" shape="general--file" />
        {requests ? requests : '...'} {t('registration requests')}
      </span>
      <span className="ecl-u-d-flex ecl-u-align-items-center">
        <Icon className="ecl-u-mr-xs" shape="general--edit" />
        {registered ? registered : '...'} {t('initiatives registered')}
      </span>
      <span className="ecl-u-d-flex ecl-u-align-items-center">
        <Icon className="ecl-u-mr-xs" shape="ui--check-filled" />
        {successful ? successful : '...'} {t('successful initiatives')}
      </span>
    </p>
  );
};

Counter.propTypes = {
  requests: PropTypes.string,
  registered: PropTypes.string,
  successful: PropTypes.string,
};

export default Counter;
