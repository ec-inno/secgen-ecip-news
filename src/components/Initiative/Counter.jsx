import React, { useState, useEffect, useContext } from 'react';
import { useTranslation } from 'react-i18next';
import axios from 'axios';

import I18nContext from '../../context/I18n';

import Icon from '../Icon';

const Counter = () => {
  const { t } = useTranslation();
  const { locale } = useContext(I18nContext);
  const { GATSBY_SITE_BASE_URL: api } = process.env;

  const [stats, setStats] = useState({});

  useEffect(() => {
    axios
      .get(`${api}/${locale}/api/counters`)
      .then(data => setStats(data))
      .catch(e => {
        console.error(t('Failed to fetch initiatives statistics'), e);
      });
  }, [locale]);

  return (
    <p className="ecl-u-type-paragraph ecl-u-mt-l ecl-u-d-flex ecl-u-justify-content-between">
      <span className="ecl-u-d-flex ecl-u-align-items-center">
        <Icon className="ecl-u-mr-xs" shape="general--file" />
        {stats.requests ? stats.requests : '...'} {t('registration requests')}
      </span>
      <span className="ecl-u-d-flex ecl-u-align-items-center">
        <Icon className="ecl-u-mr-xs" shape="general--edit" />
        {stats.registered ? stats.registered : '...'}{' '}
        {t('initiatives registered')}
      </span>
      <span className="ecl-u-d-flex ecl-u-align-items-center">
        <Icon className="ecl-u-mr-xs" shape="ui--check-filled" />
        {stats.successful ? stats.successful : '...'}{' '}
        {t('successful initiatives')}
      </span>
    </p>
  );
};

export default Counter;
