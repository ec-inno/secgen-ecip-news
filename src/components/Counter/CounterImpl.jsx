import React from 'react';

import { useI18nContext } from '@eci/context/I18n';
import { useCountersApi } from '@eci/utils/useCountersApi';

import Counter from './Counter';

const CounterImpl = () => {
  const { locale } = useI18nContext();
  const { response } = useCountersApi(locale);

  if (!response || !response.data) return '';

  return <Counter {...response.data} />;
};

export default CounterImpl;
