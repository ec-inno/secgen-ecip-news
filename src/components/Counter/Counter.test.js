import React from 'react';
import renderer from 'react-test-renderer';

import Counter from './Counter';

describe('Counter', () => {
  it('renders placeholders on lack of statistics', () => {
    const tree = renderer.create(<Counter />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders requests when provided', () => {
    const tree = renderer.create(<Counter requests="1" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders registered when provided', () => {
    const tree = renderer.create(<Counter registered="2" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders successful when provided', () => {
    const tree = renderer.create(<Counter successful="3" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
