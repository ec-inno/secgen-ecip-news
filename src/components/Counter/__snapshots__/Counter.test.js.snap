// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter renders placeholders on lack of statistics 1`] = `
<p
  className="ecl-u-type-paragraph ecl-u-mt-l ecl-u-d-flex ecl-u-justify-content-between"
>
  <span
    className="ecl-u-d-flex ecl-u-align-items-center"
  >
    <svg
      aria-hidden="true"
      className="ecl-u-mr-xs ecl-icon ecl-icon--m"
      focusable="false"
    >
      
      
      <use
        xlinkHref="file#general--file"
      />
    </svg>
    ...
     
    registration requests
  </span>
  <span
    className="ecl-u-d-flex ecl-u-align-items-center"
  >
    <svg
      aria-hidden="true"
      className="ecl-u-mr-xs ecl-icon ecl-icon--m"
      focusable="false"
    >
      
      
      <use
        xlinkHref="file#general--edit"
      />
    </svg>
    ...
     
    initiatives registered
  </span>
  <span
    className="ecl-u-d-flex ecl-u-align-items-center"
  >
    <svg
      aria-hidden="true"
      className="ecl-u-mr-xs ecl-icon ecl-icon--m"
      focusable="false"
    >
      
      
      <use
        xlinkHref="file#ui--check-filled"
      />
    </svg>
    ...
     
    successful initiatives
  </span>
</p>
`;

exports[`Counter renders registered when provided 1`] = `
<p
  className="ecl-u-type-paragraph ecl-u-mt-l ecl-u-d-flex ecl-u-justify-content-between"
>
  <span
    className="ecl-u-d-flex ecl-u-align-items-center"
  >
    <svg
      aria-hidden="true"
      className="ecl-u-mr-xs ecl-icon ecl-icon--m"
      focusable="false"
    >
      
      
      <use
        xlinkHref="file#general--file"
      />
    </svg>
    ...
     
    registration requests
  </span>
  <span
    className="ecl-u-d-flex ecl-u-align-items-center"
  >
    <svg
      aria-hidden="true"
      className="ecl-u-mr-xs ecl-icon ecl-icon--m"
      focusable="false"
    >
      
      
      <use
        xlinkHref="file#general--edit"
      />
    </svg>
    2
     
    initiatives registered
  </span>
  <span
    className="ecl-u-d-flex ecl-u-align-items-center"
  >
    <svg
      aria-hidden="true"
      className="ecl-u-mr-xs ecl-icon ecl-icon--m"
      focusable="false"
    >
      
      
      <use
        xlinkHref="file#ui--check-filled"
      />
    </svg>
    ...
     
    successful initiatives
  </span>
</p>
`;

exports[`Counter renders requests when provided 1`] = `
<p
  className="ecl-u-type-paragraph ecl-u-mt-l ecl-u-d-flex ecl-u-justify-content-between"
>
  <span
    className="ecl-u-d-flex ecl-u-align-items-center"
  >
    <svg
      aria-hidden="true"
      className="ecl-u-mr-xs ecl-icon ecl-icon--m"
      focusable="false"
    >
      
      
      <use
        xlinkHref="file#general--file"
      />
    </svg>
    1
     
    registration requests
  </span>
  <span
    className="ecl-u-d-flex ecl-u-align-items-center"
  >
    <svg
      aria-hidden="true"
      className="ecl-u-mr-xs ecl-icon ecl-icon--m"
      focusable="false"
    >
      
      
      <use
        xlinkHref="file#general--edit"
      />
    </svg>
    ...
     
    initiatives registered
  </span>
  <span
    className="ecl-u-d-flex ecl-u-align-items-center"
  >
    <svg
      aria-hidden="true"
      className="ecl-u-mr-xs ecl-icon ecl-icon--m"
      focusable="false"
    >
      
      
      <use
        xlinkHref="file#ui--check-filled"
      />
    </svg>
    ...
     
    successful initiatives
  </span>
</p>
`;

exports[`Counter renders successful when provided 1`] = `
<p
  className="ecl-u-type-paragraph ecl-u-mt-l ecl-u-d-flex ecl-u-justify-content-between"
>
  <span
    className="ecl-u-d-flex ecl-u-align-items-center"
  >
    <svg
      aria-hidden="true"
      className="ecl-u-mr-xs ecl-icon ecl-icon--m"
      focusable="false"
    >
      
      
      <use
        xlinkHref="file#general--file"
      />
    </svg>
    ...
     
    registration requests
  </span>
  <span
    className="ecl-u-d-flex ecl-u-align-items-center"
  >
    <svg
      aria-hidden="true"
      className="ecl-u-mr-xs ecl-icon ecl-icon--m"
      focusable="false"
    >
      
      
      <use
        xlinkHref="file#general--edit"
      />
    </svg>
    ...
     
    initiatives registered
  </span>
  <span
    className="ecl-u-d-flex ecl-u-align-items-center"
  >
    <svg
      aria-hidden="true"
      className="ecl-u-mr-xs ecl-icon ecl-icon--m"
      focusable="false"
    >
      
      
      <use
        xlinkHref="file#ui--check-filled"
      />
    </svg>
    3
     
    successful initiatives
  </span>
</p>
`;
